### PR TITLE
drivers: eth_sam_gmac: Use CONFIG_PTP_CLOCK_INIT_PRIORITY to init ptp

### LINKED ISSUE #67836: Sam E70 Explained gPTP does not work
Hi, this is a vanilla zephyr install from the getting started guide, with a vanilla gPTP project running on a Sam E70 Xplained board.

I have built the gPTP utility from the guide: https://docs.zephyrproject.org/latest/samples/net/gptp/README.html

The only change I have made is to configure IP addresses and add "CONFIG_NET_GPTP_LOG_LEVEL_DBG=y" into the prj.conf file.

**Describe the bug**
When trying to sync from the PC utility I get:

`
~/gptp/build$ sudo ./gptp eno1 -F gptp_cfg.ini
INFO     : GPTP [11:30:23:798] gPTP starting
ERROR    : GPTP [11:30:23:798] Group ptp not found, will try root (0) instead
INFO     : GPTP [11:30:23:798] priority1 = 248
INFO     : GPTP [11:30:23:798] announceReceiptTimeout: 3
INFO     : GPTP [11:30:23:798] syncReceiptTimeout: 3
INFO     : GPTP [11:30:23:798] LINKSPEED_100MB - PHY delay
			TX: 1044 | RX: 2133
INFO     : GPTP [11:30:23:798] LINKSPEED_1G - PHY delay
			TX: 184 | RX: 382
INFO     : GPTP [11:30:23:798] neighborPropDelayThresh: 800
INFO     : GPTP [11:30:23:798] syncReceiptThreshold: 8
INFO     : GPTP [11:30:23:798] SyncFollowUp with negative correction field: forbidden
ERROR    : GPTP [11:30:23:799] Using clock device: /dev/ptp0
STATUS   : GPTP [11:30:23:799] Starting PDelay
STATUS   : GPTP [11:30:23:799] Link Speed: 100000 kb/sec
STATUS   : GPTP [11:30:26:799] *** Announce Timeout Expired - Becoming Master
STATUS   : GPTP [11:30:26:799] New Grandmaster "E0:70:EA:FF:FE:CA:5B:6E" (previous "00:00:00:00:00:00:00:00")
EXCEPTION: GPTP [11:30:26:832] PDelay Response Receipt Timeout
EXCEPTION: GPTP [11:30:27:845] PDelay Response Receipt Timeout
EXCEPTION: GPTP [11:30:28:855] PDelay Response Receipt Timeout
EXCEPTION: GPTP [11:30:29:866] PDelay Response Receipt Timeout
EXCEPTION: GPTP [11:30:30:877] PDelay Response Receipt Timeout.........[repeats forever]`

And in the zephyr console as viewed from minicom, I get the following:
`:01:07.228,000] <inf> eth_sam: Link up
[00:01:07.329,000] <inf> net_config: IPv6 address: fe80::fec2:3dff:fe12:11ba
[00:01:07.329,000] <inf> net_config: IPv6 address: fe80::fec2:3dff:fe12:11ba
[00:01:15.143,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 0 pkt 0x2040eed0
[00:01:15.143,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:15.143,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:16.155,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 1 pkt 0x2040eed0
[00:01:16.155,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:16.155,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:17.164,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 2 pkt 0x2040eed0
[00:01:17.164,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:17.164,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:18.174,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 3 pkt 0x2040eed0
[00:01:18.174,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:18.174,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:19.185,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 4 pkt 0x2040eed0
[00:01:19.185,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:19.185,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:20.198,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 5 pkt 0x2040eed0
[00:01:20.198,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:20.198,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:21.209,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 6 pkt 0x2040eed0
[00:01:21.209,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:21.209,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:22.216,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 7 pkt 0x2040eed0
[00:01:22.216,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:22.216,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:23.218,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 8 pkt 0x2040eed0
[00:01:23.218,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:23.218,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:24.229,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 9 pkt 0x2040eed0
[00:01:24.229,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:24.229,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:25.238,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 10 pkt 0x2040eed0
[00:01:25.238,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:25.238,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
[00:01:26.252,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): Received PDELAY_REQ seq 11 pkt 0x2040eed0
[00:01:26.252,000] <dbg> net_gptp: gptp_handle_critical_msg: (rx_q[1]): No port found for gPTP buffer
[00:01:26.252,000] <dbg> net_gptp: gptp_handle_msg: (gptp): No port found for ptp buffer
`

Please also mention any information which could help others to understand
the problem you're facing:
Sam E70 Xplained Board
I stripped back to this basic setup from a bigger project and have only change the debug level as described above.
I have not gone back to any previous commits yet, but happy to do so if someone can help point me to a known good one.

**To Reproduce**
Vanilla zephyr on current release.
`west build -p always -b sam_e70_xplained samples/net/gptp/`
`west flash`

**Expected behavior**
I expected to see on the linux utility as per the documentation for the gPTP project:
`STATUS   : GPTP [13:01:25:965] AsCapable: Enabled`

**Impact**
This project was for a PTP, so a bit of a showstopper for this one.

**Environment (please complete the following information):**
Tried on Mac, and Windows

I have tried setting:
`CONFIG_NET_GPTP_NUM_PORTS=1`
and
`CONFIG_NET_GPTP_NUM_PORTS=32`
Both dont change anything.

### PR DESCRIPTION
**Pull request by talih0**
Use dedicated Kconfig symbol to set init priority.

Fixes #67836

### CODE REVIEW

**Comment by talih0** [comment]
Currently, the ptp clock is getting initialized after gptp initialization (called in net_init()), which is probably what's causing the issue in https://github.com/zephyrproject-rtos/zephyr/issues/67836

$ west build -p -b sam_e70_xplained samples/net/gptp
$ west build -t initlevels 
...
POST_KERNEL
  __init_enable_logger: enable_logger(NULL)
  __init_enable_shell_uart: enable_shell_uart(NULL)
  __init__mbedtls_init: _mbedtls_init(NULL)
  __init_malloc_prepare: malloc_prepare(NULL)
  __init_k_sys_work_q_init: k_sys_work_q_init(NULL)
  __init___device_dts_ord_24: i2c_sam_twihs_initialize(__device_dts_ord_24)
  __init___device_dts_ord_33: i2c_sam_twihs_initialize(__device_dts_ord_33)
  __init___device_dts_ord_27: mdio_sam_initialize(__device_dts_ord_27)
  __init___device_dts_ord_28: phy_mii_initialize(__device_dts_ord_28)
  __init___device_dts_ord_30: eth_initialize(__device_dts_ord_30)
  __init_net_init: net_init(NULL)
  __init___device_gmac_ptp_clock_0: ptp_gmac_init(__device_gmac_ptp_clock_0)
APPLICATION
  __init_init_app: init_app(NULL)
SMP

**Review by jukkar — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Review by decsny — COMMENTED** [review]
edit: I misread who was the maintainer of this driver, looks like nandojve is, since he has approval I removed the DNM, my bad

**Review by nashif — APPROVED** [review]

**Pull request merged**
